play-json/jvm/src/main/scala/play/api/libs/json/jackson/JacksonJson.scala

```diff
@@ -69,6 +69,16 @@ private[jackson] class JsValueSerializer(parserSettings: JsonParserSettings) ext
   import com.fasterxml.jackson.databind.node.BigIntegerNode
   import com.fasterxml.jackson.databind.node.DecimalNode
 
+  private def stripTrailingZeros(bigDec: JBigDec): JBigDec = {
+    val stripped = bigDec.stripTrailingZeros
+    if (parserSettings.bigDecimalSerializerSettings.preserveZeroDecimal && bigDec.scale > 0 && stripped.scale == 0) {
+      // restore .0 if rounded to a whole number
+      stripped.setScale(1)
```

Comment:
I don't see a straightforward way to do that with the BigDecimal API.

It looks like `scala.BigDecimal` has an `isWhole` method, but it actually calls `stripTrailingZeros` anyway (at least for the cases where this would be triggered), so it wouldn't help.

Fortunately, increasing the scale is a fairly straightforward operation.

Reply:
Seems quitd weird to me

Reply:
I don't see a reasonable way to do this with `BigDecimal` itself.

I tried multiplying by 10, then stripping zeros, then dividing by 10, but `stripTrailingZeros` will actually strip all trailing zeros, even before the decimal point.

Another option is to strip the trailing zeros from the string instead of the `BigDecimal`. I'm not opposed to that, but it seems clunkier.

Given all that, I think setting the scale is the best option.